src/linearmixedmodel.jl

```diff
@@ -408,7 +408,7 @@ function createAL(reterms::Vector{<:AbstractReMat{T}}, Xy::FeMat{T}) where {T}
             end
         end
     end
-    return A, L
+    return identity.(A), identity.(L)
```

Comment:
this will tighten up the eltype of the vector when all elements are of a single type

Reply:
It is a good idea but I don't think it will change things in the current setup. At least for `L`, `first(L)` is always `Diagonal` or `UniformBlockDiagonal` and `last{L)` is always `Matrix{T}`.

Reply:
Yeah, that matches my own sanity checks. I have some further ideas building upon this, but haven't had time to experiment more. 😄